### DISCUSSION TITLE
wire:snapshot

### DISCUSSION THREAD

**Discussion by ca-kraa** [livewire/livewire#8804] [1. Help] [answered]
Hello everyone,
I have a question about using snapshots in HTML. Here is a snippet of my code:

- Laravel 11
- Livewire 3
- PHP 8.3

```html
<div wire:snapshot='{"data":{"products":[null,{"keys":[1,2],"class":"Illuminate\\Database\\Eloquent\\Collection","modelClass":"App\\Models\\Product","s":"elcln"}],"categories":[null,{"keys":[1,2],"class":"Illuminate\\Database\\Eloquent\\Collection","modelClass":"App\\Models\\CodeProduct","s":"elcln"}],"activeTab":"all"},"memo":{"id":"1HmWmPG7HYsG1uoqSRYi","name":"badag-product","path":"\/","method":"GET","children":[],"scripts":[],"assets":[],"errors":[],"locale":"en"},"checksum":"7fba254ff8f94863967481a9b102c1a3d4f3972adbfc814a652942387a92d161"}'
    wire:effects="[]" wire:id="1HmWmPG7HYsG1uoqSRYi" class="mt-2">
```

Is it safe to display the model in HTML like this? It seems similar to using a debug bar where the model is shown. I’m confused about this and would like to know how to encrypt the snapshot.

Thank you!

**Comment by PhiloNL** [comment, marked as answer]
This is pretty safe, it only shows the namespace, so this isn't very sensitive data. If I recall correctly, you could use a morph map to hide the namespace with a single string:

```
Relation::enforceMorphMap([
    'post' => 'App\Models\Post',
    'video' => 'App\Models\Video',
]);
```

Encrypting the snapshot data object is not possible given that the front-end client relies on this information.

**Reply by ca-kraa** [reply]
I feel calm now to push to production, thank you very much. 😄
